### PR TITLE
JAPI-324 Add createsuperfile support

### PR DESCRIPTION
**Pull request by rpastrana**
Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] I have performed unit tests to cover my changes.
  - [ ] I have performed system test and covered possible regressions and side effects.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java [hunk 1]
```diff
@@ -509,6 +511,63 @@ private void checkSuperfileLayouts(String[] subfiles, String cluster) throws Exc
...
+        catch (EspSoapFault e)
+        {
+            handleEspSoapFaults(new EspSoapFaultWrapper(e), "Could Not perform createSuperfile");
+        }
+
+        if (resp.getExceptions() != null)
+            handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could Not perform createSuperfile");
+
+        return resp.getSubfiles();
+    }
```
Comment (drealeed) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsDFUClient.java:570]: can you have a method that just takes a request wrapper and returns a response wrapper?  This way the existingFIle flag is available. And if more stuff gets added down the road we won't have to change the interface.

File: wsclient/src/test/java/org/hpccsystems/ws/client/WsDFUClientTest.java [hunk 2]
```diff
@@ -378,4 +379,21 @@ public void getClustersTest() throws Exception
...
     }
 
+    @Test
+    public void createSuperFileTest() throws Exception
+    {
+        try
+        {
+            wsdfuclient.createSuperfile("hpcc4j::tests::junit::mysuperfile", new String [] {""});
+        }
+        catch (Exception e)
```
Comment (drealeed) [inline wsclient/src/test/java/org/hpccsystems/ws/client/WsDFUClientTest.java:389]: can the test exercise test both creating a nonexistent superfile and adding subfiles to an existing superfile, and trying to add a nonexistent subfile to check the error handling?

**Comment by rpastrana** [comment]
@drealeed please review

**Comment by rpastrana** [comment]
Thanks @drealeed 
@jpmcmu please review.

**Comment by jpmcmu** [comment]
@rpastrana looks good to me

**Pull request merged**